node_modules/oae-transfer/lib/internal/dao.js

```diff
@@ -0,0 +1,778 @@
+    }
+
+    // Insert to library 
+    _insertOnLibrary(resource, principalId, constant, function(err) {
+        if(err) {
+            return callback(err);
+        }
+
+        // Remove to library 
+        _removeFromLibrary(resource, principalIdOrigin, constant, function(err) {
+            if(err) {
+                return callback(err);
+            }
+            return callback(null);                
+        }); 
+    }); 
+};
+
+/**
+ * Get the name of the resource by it's id 
+ * take advantage of the differentiation of data type to modify the library
+ * 
+ * @param {Object} ctx
+ * @param {String} idResource
+ *
+ * @returns resource
+ * @api private
+ */
+var _getDocument = function (ctx, idResource, callback) {
+    var resourceType = idResource.split(":")[0];
+    // If the resource is a Content
+    if(resourceType === "c") {     
+        ContentDAO.getContent(idResource, function(err, result) {
+            if(err) {
+                return callback(err);
+            }
+            return callback(null, result);        
+        });
+
+    // If the resource is a Folder
+    } else if (resourceType === "g") {
+        _getFoldersByGroupIds(ctx, idResource, function(err, result) {
+            // if err == 1 the resource is a group
+            if(err) {
+                if(err === 1) {
+                    return callback(null, result);
+                }
+                return callback(err);
+            }
+            return callback(null, result);                
+        }); 
+
+    // If the resource is a Discussion
+    } else if (resourceType === "d") {
+        DiscussionsDAO.getDiscussion(idResource, function(err, result) {
+            if(err) {
+                return callback(err);
+            }
+            return callback(null, result);                
+        }); 
+
+    // If the resource is a Meeting
+    } else if (resourceType === "m") {
+        MeetingsDAO.getMeeting(idResource, function(err, result) {
+            if(err) {
+                return callback(err);
+            }
+            return callback(null, result);                
+        }); 
+    }
+};
+
+/**
+ * Get folder by group id
+ * 
+ * @param {Object} ctx
+ * @param {String} idResource
+ *
+ * @returns folder
+ * @api private
+ */
+ var _getFoldersByGroupIds = function (ctx, idResource, callback) {
+    Cassandra.runQuery('SELECT * FROM "FoldersGroupId" WHERE "groupId" = ?', [idResource], function(err, result) {
+        
+        // The resource is a groupe if there is no result to this query
+        if(_.chain(result).map(Cassandra.rowToHash)._wrapped[0] === undefined) {
+            GroupAPI.getGroup(ctx, idResource, function(err, group) {
+                if(err) {
+                    return callback(err);
+                }
+                return callback(1, group);
+            });
+        } else {
+            FoldersDAO.getFolder(_.chain(result).map(Cassandra.rowToHash)._wrapped[0].folderId, function(err, folder) {
```

Comment:
This might be the first time chaining actually makes sense, but it should be something like `_.chain(result).map(Cassandra.rowToHash).pluck('folderId').value()`.